announcement/models/announcement.py

```diff
@@ -84,6 +85,16 @@ def _compute_allowed_user_ids(self):
             announcement.allowed_user_ids = announcement.user_group_ids.users
             announcement.allowed_users_count = len(announcement.user_group_ids.users)
 
+    @api.onchange("is_general_announcement")
+    def _onchange_is_general_announcement(self):
+        # If the option "General announcement" is checked, we assign the group
+        # "Internal users" to the user_group_ids field.
+        general_announcement = self.filtered(lambda x: x.is_general_announcement)
```

Comment:
You don't need to loop over self. The onchange decorator is always triggered over a singleton